src/java/org/apache/cassandra/db/MultiCBuilder.java

```diff
@@ -193,6 +194,9 @@ public NavigableSet<Clustering<?>> build()
         if (clusterings.isEmpty())
             return BTreeSet.of(comparator, Clustering.EMPTY);
 
+        if (clusterings.size() == 1)
```

Comment:
Nit: Maybe adding a short comment on what this optimization implies?

Reply:
Good idea. In short, this is a very hot path that both read and write paths run through. Avoiding builder and iterator allocations that would otherwise happen below is useful.

Reply:
Thanks! I'm +1 (non binding) when that comment is added.

Also, I guess this also needs an update to `CHANGES.TXT`?